### PR TITLE
chore(workflows-sdk): add idempotencyKey to shared context

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**

### CODE REVIEW

**Review by adrien2p — APPROVED** [review]
🔥

File: packages/workflows-sdk/src/utils/composer/create-step.ts [hunk 1]
```diff
@@ -121,6 +121,9 @@ function applyStep<
 
     const handler = {
       invoke: async (transactionContext) => {
+        transactionContext.context.idempotencyKey =
+          transactionContext.idempotency_key
```
Comment (srindom) [inline packages/workflows-sdk/src/utils/composer/create-step.ts]: Shouldn't it be?
```
transactionContext.metadata.idempotency_key
```

Comment (adrien2p) [reply packages/workflows-sdk/src/utils/composer/create-step.ts]: It is in too many places, but it is also in the transaction context i believe 😅

Comment (carlos-r-l-rodrigues) [reply packages/workflows-sdk/src/utils/composer/create-step.ts]: I added the correct type to the handler.

**Pull request merged**